cfme/tests/storage/test_volume.py

```diff
@@ -92,29 +134,62 @@ def test_storage_volume_crud(appliance, provider):
         casecomponent: Cloud
     """
     # create volume
-    volume_collection = appliance.collections.volumes
-    manager_name = '{} Cinder Manager'.format(provider.name)
-    volume = volume_collection.create(name=fauxfactory.gen_alpha(),
-                                      storage_manager=manager_name,
-                                      tenant=provider.data['provisioning']['cloud_tenant'],
-                                      size=STORAGE_SIZE,
-                                      provider=provider)
-    assert volume.exists
+    volume = create_volume(appliance, provider, from_manager, should_assert=True)
 
     # update volume
     old_name = volume.name
     new_name = fauxfactory.gen_alpha()
-    with update(volume):
-        volume.name = new_name
-
-    with update(volume):
-        volume.name = old_name
+    updates = {'volume_name': new_name, 'volume_size': STORAGE_SIZE + 1}
+    if provider.one_of(OpenStackProvider):
+        updates = {'volume_name': new_name}
+
+    if from_manager:
+        storage_manager = appliance.collections.block_managers.filter(
+            {"provider": provider}).all()[0]
+    else:
+        storage_manager = None
+    volume = volume.update(updates, storage_manager)
```

Comment:
This should be called through the `Updateable` context manger, like it was before. Then you don't have to instantiate a new volume object and return it. It gets modified in place within the context manager.

Reply:
I would have to add size as argument. And size would have to be set even in creation. That didn't make sense for me and @digitronik so I reworked it like this.

Reply:
@mmojzis you are refactored most of part for this PR. If you resolve conflict between size as attribute and property then it will be good. I would like to resolve this conflict if you ack.

Reply:
Also I have to use volume.update because there are multiple possible navigations specified.